sentry/src/test/java/io/sentry/BaggageTest.kt

```diff
@@ -0,0 +1,649 @@
+            if (i <= MAX_BAGGAGE_LIST_MEMBER_COUNT) {
+                expectedItems.add("a$key=$i")
+            }
+        }
+
+        val expectedHeaderString = expectedItems.joinToString(",")
+        assertEquals(expectedHeaderString, baggage.toHeaderString())
+    }
+
+    @Test
+    fun `null value is omitted from header string`() {
+        val baggage = Baggage(logger)
+
+        baggage.setTraceId(null)
+        baggage.setPublicKey(null)
+        baggage.setRelease(null)
+        baggage.setEnvironment(null)
+        baggage.setTransaction(null)
+        baggage.setUserId(null)
+        baggage.setUserSegment(null)
+
+        assertEquals("", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `can set values from trace context`() {
+        val baggage = Baggage(logger)
+        val publicKey = Dsn(dsnString).publicKey
+        val traceId = SentryId().toString()
+        val userId = UUID.randomUUID().toString()
+
+        baggage.setTraceId(traceId)
+        baggage.setPublicKey(publicKey)
+        baggage.setRelease("1.0-rc.1")
+        baggage.setEnvironment("production")
+        baggage.setTransaction("TX")
+        baggage.setUserId(userId)
+        baggage.setUserSegment("segmentA")
+
+        assertEquals("sentry-environment=production,sentry-publickey=$publicKey,sentry-release=1.0-rc.1,sentry-traceid=$traceId,sentry-transaction=TX,sentry-userid=$userId,sentry-usersegment=segmentA", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `duplicate entries are lost`() {
+        val baggage = Baggage.fromHeader("duplicate=a,duplicate=b", logger)
+        assertEquals("duplicate=b", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `setting a value multiple times only keeps the last`() {
+        val baggage = Baggage.fromHeader("sentry-traceid=a", logger)
+
+        baggage.setTraceId("b")
+        baggage.setTraceId("c")
+
+        assertEquals("sentry-traceid=c", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `value may contain = sign`() {
+        val baggage = Baggage(logger)
+
+        baggage.setTransaction("a=b")
+
+        assertEquals("sentry-transaction=a%3Db", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `corrupted string does not throw out`() {
+        val baggage = Baggage.fromHeader("a", logger)
+        assertEquals("", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `corrupted string does not throw out 2`() {
+        val baggage = Baggage.fromHeader("a=b=", logger)
+        assertEquals("", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `corrupted string can be parsed partially`() {
+        val baggage = Baggage.fromHeader("a=value,b", logger)
+        assertEquals("a=value", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `baggage value encoding`() {
+        // keep baggage-octet: %x21 / %x23-2B / %x2D-3A / %x3C-5B / %x5D-7E
+        val values = percentEncodedValues().also {
+//            it["!"] = "!" // instead of %21
```

Comment:
Why those are commented out?

Reply:
> Characters which are not required to be percent-encoded MAY be percent-encoded.

I added them to the test but the default implementation on Java doesn't encode them. Since we don't have to encode them I commented them out but left them there to have an overview of what's happening.

Want me to remove them?

Reply:
Can you drop a comment about the Java implementation somewhere (maybe here in the test)?

Reply:
Added a comment